File: 0mq/PUBSUB/vector_client.py
import numpy as np # pkg load signal;
import zmq         # pkg load zeromq;
import array
from matplotlib import pyplot as plt

Nt=256
context=zmq.Context()

sock1=context.socket(zmq.SUB) # sock1=zmq_socket(ZMQ_SUB);
sock1.connect("tcp://127.0.0.1:5556");
sock1.setsockopt(zmq.SUBSCRIBE, b"")
vector1=[]
while (len(vector1)<Nt):
  raw_recv=sock1.recv()         
#  recv=array.array('f',raw_recv)
  recv=array.array('i',raw_recv) 
  print(f"{recv[0]} {recv[1]} {recv[-1]}")
  plt.plot(recv)
  plt.show()
#    vector1tmp=recv[0::2]
#    vector2tmp=recv[1::2]

File: 0mq/PUBSUB/vector_client.m
pkg load zeromq;
Nt=2*1024
sock1=zmq_socket(ZMQ_SUB);
zmq_connect(sock1,"tcp://127.0.0.1:5556");
zmq_setsockopt(sock1, ZMQ_SUBSCRIBE, "");
do
  recv=zmq_recv(sock1, Nt*8, 0);
% vector=typecast(recv,"single complex");
  vector=typecast(recv,"int32");
  plot(vector)
  refresh()
  printf("%d %d %d\n",vector(1), vector(2),vector(end))
until (1==0)
